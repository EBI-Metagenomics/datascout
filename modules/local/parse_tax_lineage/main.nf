process TAX_LINEAGE {

    container 'quay.io/biocontainers/ete3:3.1.2'

    label "process_medium"

    tag "${meta}"

    input:
      tuple val(meta), val(taxid)
      val(taxdump)
      val(db_path)

    output:
      tuple val(meta), path("*_tax_ranks.tsv"), emit: tax_ranks
      path("versions.yml"), emit: versions

    script:
    prefix = meta.id
    """
    cp -r ${db_path} ./
    parse_tax_lineage.py --taxid ${taxid} --output ${prefix}_tax_ranks.tsv --taxdump "${taxdump}" --db_path "${db_path}"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
    \$( parse_tax_lineage.py --taxdump "${taxdump}" --db_path "${db_path}" --version 2>&1 )
      Python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """
}

// Get taxonomic lineage and ranks of query genome from NCBI
