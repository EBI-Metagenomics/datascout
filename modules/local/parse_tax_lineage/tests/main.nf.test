nextflow_process {

    name "Test Process TAX_LINEAGE"
    script "modules/local/parse_tax_lineage/main.nf"
    process "TAX_LINEAGE"

    test("Should run without failures") {


        setup {
            "${projectDir}/tests/reference_databases/ncbi/make_mini_db.py".execute().waitFor()
        }

        when {
            process {
                """
                input[0] = tuple([id:"test_sample1", genome_id:"GCA_003719455.1", ena_tax:"default"], 5693)
                input[1] = "${projectDir}/tests/reference_databases/ncbi/mini_taxdump.tar.gz"
                input[2] = "${projectDir}/tests/reference_databases/ncbi/mini_trypanosoma.sqlite"
                """
            }
        }


        then {
            assertAll{
                { assert process.success }
                { assert snapshot(
                    process.out.tax_ranks,
                    process.out.versions
                ).match()   }
            }
        }
    }
}