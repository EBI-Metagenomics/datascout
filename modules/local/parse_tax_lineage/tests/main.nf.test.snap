{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "test_sample1",
                        "genome_id": "GCA_003719455.1",
                        "ena_tax": "default"
                    },
                    "test_sample1_tax_ranks.tsv:md5,b06eebb92f06f93c2894fb5f4e685d13"
                ]
            ],
            [
                "versions.yml:md5,b000fb6960a0415110097b73d9a673fe"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-11T10:52:41.859143"
    }
}